docs/add-a-new-datasource.md

```diff
@@ -32,6 +32,10 @@ Attribute names are to specified in `snake_case` as opposed to the AWS API which
 
 These will map the AWS API response to the data source schema. You will also need to handle different response types (including errors correctly). For complex attributes you will need to implement Flattener or Expander functions. The [Data Handling and Conversion Guide](data-handling-and-conversion.md) covers everything you need to know for mapping AWS API responses to Terraform State and vice-versa. The [Error Handling Guide](error-handling.md) covers everything you need to know about handling AWS API responses consistently.
 
+### Register Data Source to the provider
+
+Data Sources use a self registration process that adds them to the provider using the `@SDKDataSource()` annotation in the datasource's comments. Run `make servicepackages` to register the datasource. This will add an entry to the `service_package_gen.go` file located in the service package folder.
```

Comment:
Maybe include a code snippet demonstrating the comment in context?

Reply:
added an example to both files.